crates/hyperdrive-math/src/short/max.rs

```diff
@@ -569,14 +601,36 @@ mod tests {
                 .await
             {
                 Ok(expected) => {
-                    // TODO: remove this tolerance when calculate_open_short
-                    // rust implementation matches solidity.
-                    // Currently, only about 1 - 4 / 1000 tests aren't
-                    // exact matchces. Related issue:
-                    // https://github.com/delvtech/hyperdrive-rs/issues/45
-                    assert_eq!(
-                        U256::from(actual.unwrap().unwrap()) / uint256!(1e12),
-                        expected / uint256!(1e12)
+                    // Make sure the solidity & rust runctions gave the same value.
+                    let actual_unwrapped = actual.unwrap().unwrap();
+                    let expected_fp = FixedPoint::from(expected);
+                    let error = if expected_fp > actual_unwrapped {
+                        expected_fp - actual_unwrapped
+                    } else {
+                        actual_unwrapped - expected_fp
+                    };
+                    assert!(
+                        error < sol_correctness_tolerance,
+                        "error {} exceeds tolerance of {}",
+                        error,
+                        sol_correctness_tolerance,
+                    );
+
+                    // Make sure the pool was drained.
+                    let effective_shares = state.effective_share_reserves()?;
+                    let min_shares = state.minimum_share_reserves();
+                    assert!(effective_shares >= min_shares,
+                        "effective_share_reserves={} should always be greater than the minimum_share_reserves={}.",
+                        state.effective_share_reserves()?,
+                        state.minimum_share_reserves()
+                    );
+                    let reserve_amount_above_minimum = effective_shares - min_shares;
+                    assert!(reserve_amount_above_minimum < reserves_drained_tolerance,
+                        "effective_share_reserves={} - minimum_share_reserves={} (diff={}) should be < tolerance={}",
+                        effective_shares,
+                        min_shares,
+                        reserve_amount_above_minimum,
+                        reserves_drained_tolerance,
                     );
```

Comment:
Ok, I think that one reason we may be having problems here is if the `share_reserves < effective_share_reserves` then the `share_reserves` will be equal to the minimum share reserves, but the effective share reserves will be greater (potentially by a lot). Since this test fuzzes over general pool states, I wouldn't be surprised if this is causing the issue.

We should be able to lower the tolerance if we check the minimum share reserves against `min(share_reserves, effective_share_reserves)`.

Reply:
That did not result in an improved tolerance, unfortunately. I am going to merge, with the idea that we can focus more on understanding what's going on in my next PR fixing open short.